### PR TITLE
Update to Docusaurus 3

### LINKED ISSUE #443: Update docusaurus to latest version
Update available 2.1.0 → 2.4.1

To upgrade Docusaurus packages with the latest version, run the following command:

~~~sh
npm i @docusaurus/core@latest @docusaurus/preset-classic@latest @docusaurus/module-type-aliases@latest
~~~

### PR DESCRIPTION
**Pull request by XVincentX**
1. Update Docusaurus and dependencies
2. Update MD files to render correctly with MDX 3
3. Update Typescript to version 5
4. Removed `import React` since Docusaurus 3 uses runtime detection
5. Update devcontainer so that it works for real

### CODE REVIEW

**Comment by ralfhandl** [comment]
Update available 3.0.0 → 3.0.1

Published 20 hours ago 😄

**Comment by XVincentX** [comment]
Can we merge this in meantime? We need at least a baseline working on 3.0 - then we can move on to other versions

File: tools/api-designer/package.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
   },
   "scripts": {
     "# postinstall": "patch-package",
-    "build": "rimraf dist && parcel build --dist-dir ./dist/api-designer",
+    "prepublish": "rimraf dist",
```
Comment (ralfhandl) [inline tools/api-designer/package.json:29]: Why not `prebuild`?

Comment (XVincentX) [reply tools/api-designer/package.json:29]: I do not know :) - I just took what it was in there and separated into two commands. I can refresh this repository a little bit but I wanted to touch as little as possible for now

**Review by ralfhandl — APPROVED** [review]

File: tools/api-designer/package.json (hunk 1 above)
Comment (ralfhandl) [reply tools/api-designer/package.json:29]: In other places you used "prebuild".

**Pull request merged**